coordinator/src/services/validation/bundle-validator.ts

```diff
@@ -59,102 +27,85 @@ function verifyCancellationBundle(bundle: fhir.Bundle): Array<errors.ValidationE
-      "dispenseRequest.extension (performer site type)",
-      (medicationRequest) => getExtensionForUrl(
+  const consistencyValidators: Array<medicationRequestValidator> = [
+    {
+      fieldName: "groupIdentifier",
+      fieldAccessor: (medicationRequest) => medicationRequest.groupIdentifier
+    },
+    {
+      fieldName: "category",
+      fieldAccessor: (medicationRequest) => medicationRequest.category
+    },
+    {
+      fieldName: "authoredOn",
+      fieldAccessor: (medicationRequest) => medicationRequest.authoredOn
+    },
+    {
+      fieldName: "subject",
+      fieldAccessor: (medicationRequest) => medicationRequest.subject
+    },
+    {
+      fieldName: "requester",
+      fieldAccessor: (medicationRequest) => medicationRequest.requester
+    },
+    {
+      fieldName: "dispenseRequest.performer",
+      fieldAccessor: (medicationRequest) => medicationRequest.dispenseRequest.performer
+    },
+    {
+      fieldName: "dispenseRequest.validityPeriod",
+      fieldAccessor: (medicationRequest) => medicationRequest.dispenseRequest.validityPeriod
+    },
+    {
+      fieldName: "dispenseRequest.expectedSupplyDuration",
+      fieldAccessor: (medicationRequest) => medicationRequest.dispenseRequest.expectedSupplyDuration
+    },
+    {
+      fieldName: 'dispenseRequest.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")',
+      fieldAccessor:(medicationRequest) => getExtensionForUrl(
         medicationRequest.dispenseRequest.extension,
         "https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType",
         "MedicationRequest.dispenseRequest.extension"
       )
-    ),
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
-      medicationRequests,
-      "extension (prescription type)",
-      (medicationRequest) => getExtensionForUrl(
+    },
+    {
+      fieldName: 'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-prescriptionType")',
+      fieldAccessor: (medicationRequest) => getExtensionForUrl(
         medicationRequest.extension,
         "https://fhir.nhs.uk/R4/StructureDefinition/Extension-prescriptionType",
         "MedicationRequest.extension"
       )
-    ),
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
-      medicationRequests,
-      "extension (responsible practitioner)",
-      (medicationRequest) => getExtensionForUrlOrNull(
+    },
+    {
+      fieldName: 'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner")',
+      fieldAccessor: (medicationRequest) => getExtensionForUrlOrNull(
         medicationRequest.extension,
         "https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner",
         "MedicationRequest.extension"
       )
-    ),
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
```

Comment:
Although this one went missing

Reply:
Not sure how that happened, I'll add it in now